Cognite.Simulator.Tests/UtilsTests/TestUtilities.cs

```diff
@@ -0,0 +1,183 @@
+            var pipelineId = SeedData.TestExtPipelineId;
+            string directory = Directory.GetCurrentDirectory();
+            string filePath = Path.Combine(directory, "config.yml");
+            string yamlContent = $@"
+                version: 1
+                logger:
+                    console:
+                        level: ""debug""
+                    remote:
+                        level: ""information""
+                        enabled: true
+                cognite:
+                    project:  {project}
+                    host: {host}
+                    extraction-pipeline:
+                        pipeline-id: {pipelineId}
+                simulator:
+                    name: {simulatorName}
+                    data-set-id: {datasetId}
+
+                connector:
+                    status-interval: 3
+                    name-prefix: {SeedData.TestIntegrationExternalId}
+                    add-machine-name-suffix: false
+                    model-library:
+                        library-update-interval: 1";
+
+            // Write the content to the file
+            File.WriteAllText(filePath, yamlContent);
+        }
+
+        private static HttpResponseMessage ForbiddenResponse =
+            CreateResponse(HttpStatusCode.Forbidden, "{\"error\": {\"code\": 403,\"message\": \"Forbidden\"}}");
+
+        /// <summary>
+        /// Mocks the HttpClientFactory to return the mocked responses
+        /// Example format for endpointMappings:
+        ///     { uri => uri.Contains("/extpipes"), (MockExtPipesEndpoint, 0, 2) },
+        ///     2 is the number of times the response function will be called before returning a 403 Forbidden
+        ///     if maxCalls is null, the response function will return the same response indefinitely
+        /// </summary>
+        /// <param name="endpointMappings">Dictionary of URL matchers and response functions</param>
+        public static Func<HttpRequestMessage, CancellationToken, Task<HttpResponseMessage>> mockRequestsAsync(Dictionary<Func<string, bool>, (Func<HttpResponseMessage> responseFunc, int callCount, int? maxCalls)> endpointMappings)
+        {
+            
+            return async (HttpRequestMessage message, CancellationToken token) =>
+            {
+                var uri = message.RequestUri?.ToString();
+                if (uri == null)
+                {
+                    throw new ArgumentNullException(nameof(uri));
+                }
+
+                foreach (var mapping in endpointMappings)
+                {
+                    if (mapping.Key(uri))
+                    {
+                        var (responseFunc, callCount, maxCalls) = mapping.Value;
+                        if (maxCalls.HasValue) {
+                            if (callCount < maxCalls)
+                            {
+                                endpointMappings[mapping.Key] = (responseFunc, callCount + 1, maxCalls);
```

Comment:
We could make this thread-safe when tests run concurrently by making this a `ConcurrentDictionary`

Reply:
this is used for the tests (we run them in one thread, because there will be more issues if we don't)

Reply:
yeah, let's try that